eis/src/main/java/at/ac/tgm/controller/TeacherAmpelController.java

```diff
@@ -0,0 +1,183 @@
+        return ResponseEntity.ok(teacherAmpelService.getAllAmpelForTeacher(teacher.getId()));
+    }
+
+    @Secured(Roles.LEHRER)
+    @PostMapping
+    public ResponseEntity<?> createAmpelForTeacher(@RequestBody AmpelRequestDto dto, Authentication authentication) {
+        String sAMAccountName = authentication.getName();
+
+        if (sAMAccountName == null || sAMAccountName.isEmpty()) {
+            return ResponseEntity.status(401).body(new ErrorResponseDto("Nicht authentifiziert.", 401));
+        }
+
+
+        Optional<Teacher> teacherOptional = teacherAmpelService.getTeacherBySAMAccountName(sAMAccountName);
+        if (teacherOptional.isEmpty()) {
+            return ResponseEntity.status(404).body(new ErrorResponseDto("Lehrer in der Datenbank nicht gefunden.", 404));
+        }
+
+        Teacher teacher = teacherOptional.get();
+
+        dto.setTeacherId(teacher.getId());
+
+        try {
+            AmpelDto createdAmpel = teacherAmpelService.createOrUpdateAmpel(dto);
+            return ResponseEntity.status(201).body(createdAmpel);
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.status(400).body(new ErrorResponseDto(e.getMessage(), 400));
+        } catch (Exception e) {
+            return ResponseEntity.status(500).body(new ErrorResponseDto("Interner Serverfehler.", 500));
+        }
+    }
+
+    /**
+     * Ändert einen bestehenden Ampel-Eintrag für den aktuell authentifizierten Lehrer.
+     */
+    @Secured(Roles.LEHRER)
+    @PutMapping
+    public ResponseEntity<?> updateAmpelForTeacher(@RequestBody AmpelRequestDto dto, Authentication authentication) {
+        String sAMAccountName = authentication.getName();
+
+        if (sAMAccountName == null || sAMAccountName.isEmpty()) {
+            return ResponseEntity.status(401).body(new ErrorResponseDto("Nicht authentifiziert.", 401));
+        }
+
+        Optional<Teacher> teacherOptional = teacherAmpelService.getTeacherBySAMAccountName(sAMAccountName);
+        if (teacherOptional.isEmpty()) {
+            return ResponseEntity.status(404).body(new ErrorResponseDto("Lehrer in der Datenbank nicht gefunden.", 404));
+        }
+
+        Teacher teacher = teacherOptional.get();
+
+        if (dto.getTeacherId() == null || !dto.getTeacherId().equals(teacher.getId())) {
+            return ResponseEntity.status(403).body(new ErrorResponseDto("Zugriff auf diesen Ampel-Eintrag verweigert.", 403));
+        }
+
+        try {
+            AmpelDto updatedAmpel = teacherAmpelService.createOrUpdateAmpel(dto);
+            return ResponseEntity.ok(updatedAmpel); // 200 OK
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.status(400).body(new ErrorResponseDto(e.getMessage(), 400)); // Bad Request
+        } catch (Exception e) {
+            return ResponseEntity.status(500).body(new ErrorResponseDto("Interner Serverfehler.", 500)); // Internal Server Error
+        }
+    }
+    @Secured(Roles.LEHRER)
```

Comment:
```suggestion
    @Secured(Roles.TEACHER)
```
Bitte einfach überall mit Suchen Ersetzen korrigieren.